### ISSUE TITLE
Failed to install bap through opam on 4.02.1 & Ubuntu 14.04

### ISSUE THREAD

**Issue by zhaoguixu** [BinaryAnalysisPlatform/bap#412]
Hi,
I try to install bap through opam following the installation guide. But I always get the compiling error of   pa_sexp_conv.

```
#=== ERROR while installing pa_sexp_conv.113.00.01 ============================#
# opam-version 1.2.2
# os           linux
# command      make
# path         /home/gui/.opam/4.02.1/build/pa_sexp_conv.113.00.01
# compiler     4.02.1
# exit-code    2
# env-file     /home/gui/.opam/4.02.1/build/pa_sexp_conv.113.00.01/pa_sexp_conv-23275-c743ac.env
# stdout-file  /home/gui/.opam/4.02.1/build/pa_sexp_conv.113.00.01/pa_sexp_conv-23275-c743ac.out
# stderr-file  /home/gui/.opam/4.02.1/build/pa_sexp_conv.113.00.01/pa_sexp_conv-23275-c743ac.err
### stdout ###
# [...]
# ./setup.exe -build 
# /home/gui/.opam/4.02.1/bin/ocamlopt.opt unix.cmxa -I /home/gui/.opam/4.02.1/lib/ocaml/ocamlbuild /home/gui/.opam/4.02.1/lib/ocaml/ocamlbuild/ocamlbuildlib.cmxa myocamlbuild.ml /home/gui/.opam/4.02.1/lib/ocaml/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamldep -syntax camlp4o -package camlp4.extend -package camlp4.quotations -package type_conv -modules syntax/pa_sexp_conv.mli > syntax/pa_sexp_conv.mli.depends
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamlc -c -g -annot -bin-annot -syntax camlp4o -package camlp4.extend -package camlp4.quotations -package type_conv -I syntax -o syntax/pa_sexp_conv.cmi syntax/pa_sexp_conv.mli
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamldep -syntax camlp4o -package camlp4.extend -package camlp4.quotations -package type_conv -syntax camlp4o -modules syntax/pa_sexp_conv.ml > syntax/pa_sexp_conv.ml.depends
# /home/gui/.opam/4.02.1/bin/ocamlfind ocamlc -c -g -annot -bin-annot -syntax camlp4o -package camlp4.extend -package camlp4.quotations -package type_conv -syntax camlp4o -I syntax -o syntax/pa_sexp_conv.cmo syntax/pa_sexp_conv.ml
# + /home/gui/.opam/4.02.1/bin/ocamlfind ocamlc -c -g -annot -bin-annot -syntax camlp4o -package camlp4.extend -package camlp4.quotations -package type_conv -syntax camlp4o -I syntax -o syntax/pa_sexp_conv.cmo syntax/pa_sexp_conv.ml
# File "syntax/pa_sexp_conv.ml", line 1630, characters 62-80:
# Error: Unbound value Gen.string_of_ctyp
# Command exited with code 2.
### stderr ###
# E: Failure("Command ''/home/gui/.opam/4.02.1/bin/ocamlbuild' syntax/pa_sexp_conv.cma syntax/pa_sexp_conv.cmxa syntax/pa_sexp_conv.a syntax/pa_sexp_conv.cmxs -tag debug' terminated with error code 10")
# make: *** [build] Error 1



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions were aborted
  ∗  install bap     0.9.5 
  ∗  install cohttp  0.19.3
  ∗  install conduit 0.11.0
The following actions failed
  ∗  install pa_sexp_conv 113.00.01
...
```

Do you know what's wrong with it?

Thanks.

**Comment by ivg** [comment]
According to the error report, you're installing a very old version of bap. To install the latest release bap, do
```
opam install bap.0.9.9
```

to install the latest alpha version of bap (1.0.0~alpha):
```
opam repository add git://github.com/BinaryAnalysisPlatform/opam-repository.git
opam install bap
```

**Comment by zhaoguixu** [comment]
I have tried 0.9.9, but the issue still exists. I have checked the dependencies. opam gets a solution and will try to install type_conv.112.01.01, which is not compatible with pa_sexp_conv.113.00.01. Eventually, according to the compilation error, I download the source of type_conv.112.01.01 and fix it, and use ```opam pin add type_conv.112.01.01 <local source folder>``` to make opam points to local repos. This may not be the perfect solution, but it works. Hope it may help others having the same issue.
